### PR TITLE
Fix push error when repository is empty

### PR DESCRIPTION
**Pull request by BrainBalaka**
## Summary
When we use an empty repository url, the push task will throw an error because it cannot find the base branch (`master`). This PR fixes this by initializing a master branch if it's not yet exist, and push it first into the remote repo so it can be treated as default branch

### CODE REVIEW

File: src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs [hunk 1]
```diff
@@ -126,6 +139,24 @@ public Task<string> UpdateMigrationScript()
             return _mainProjectGenerator.UpdateMigrationScript();
         }
 
+        public async Task InitializeRepository()
```
Comment (frandi) [inline src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs]: I don't think we need it here. Doing it in GitHub plugin is enough.

File: src/Plugins/RepositoryProvider/Polyrific.Catapult.Plugins.GitHub/src/GitHubUtils.cs [hunk 2]
```diff
@@ -213,15 +178,46 @@ public Task<bool> CheckoutBranch(string localRepository, string branch)
...
+
+                    repo.Commit("Initial commit", signature, signature);
+                    var masterBranch = repo.Branches["master"];
+
+                    var remote = repo.Network.Remotes["origin"];
+                    repo.Branches.Update(masterBranch,
+                        b => b.Remote = remote.Name,
+                        b => b.UpstreamBranch = masterBranch.CanonicalName);
+
+                    await Push(remoteUrl, localRepository, baseBranch);
```
Comment (frandi) [inline src/Plugins/RepositoryProvider/Polyrific.Catapult.Plugins.GitHub/src/GitHubUtils.cs]: No, don't do it here. Please do `Push` action in `Push` method instead.

File: src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs (hunk 1 above)
Comment (BrainBalaka) [reply src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs]: Probably the method name is misleading. This method would create `readme.md` file and `.gitignore` which I think can be different for each generator, we would not do something like `git init` here. I would rename the method to `CreateRepositoryFiles()` instead.

File: src/Plugins/RepositoryProvider/Polyrific.Catapult.Plugins.GitHub/src/GitHubUtils.cs (hunk 2 above)
Comment (BrainBalaka) [reply src/Plugins/RepositoryProvider/Polyrific.Catapult.Plugins.GitHub/src/GitHubUtils.cs]: @frandi I've moved the Push master code into Push method.

File: src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs (hunk 1 above)
Comment (frandi) [reply src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs]: ok sure.

Comment (BrainBalaka) [reply src/Plugins/GeneratorProvider/Polyrific.Catapult.Plugins.AspNetCoreMvc/src/CodeGenerator.cs]: Almost forgot about this one :) anyway, I've renamed into `GenerateRepositoryFiles()`

**Pull request merged**